### PR TITLE
Reduce nuget spew

### PR DESCRIPTION
**Pull request by KevinRansom**
Simplify the #r "nuget:"  console output -- now tells you only what was requested is output.

### CODE REVIEW

File: src/Microsoft.DotNet.Interactive.PackageManagement/KernelSupportsNugetExtensions.cs
Comment (jonsequitur) [inline src/Microsoft.DotNet.Interactive.PackageManagement/KernelSupportsNugetExtensions.cs:228]: Why `Contains` instead of `Equals`? This looks vulnerable to false matches, e.g. `Plotly.NET.Interactive` contains `Plotly.NET`.

**Comment by jonsequitur** [comment]
Can we add a test that pins the intended behavior?

File: src/Microsoft.DotNet.Interactive.PackageManagement/KernelSupportsNugetExtensions.cs
Comment (KevinRansom) [reply src/Microsoft.DotNet.Interactive.PackageManagement/KernelSupportsNugetExtensions.cs:228]: I want to add some logging too.

Comment (KevinRansom) [reply src/Microsoft.DotNet.Interactive.PackageManagement/KernelSupportsNugetExtensions.cs:228]: When I wrote it I think I was worried that the names contained version.  It turns out they are just PackageNames so equals will be fine.

Comment (KevinRansom) [reply src/Microsoft.DotNet.Interactive.PackageManagement/KernelSupportsNugetExtensions.cs:228]: Brain-fart
--- requested is a list of PackageNames so, it succeeds when the list contains the exact match modulo StringComparison.OrdinalIgnoreCase.

````
 var requested =
          kernel.RequestedPackageReferences
                .Except(kernel.ResolvedPackageReferences, PackageReferenceComparer.Instance)
                .Select(s => s.PackageName).OrderBy(s => s).ToList();
````

Comment (KevinRansom) [reply src/Microsoft.DotNet.Interactive.PackageManagement/KernelSupportsNugetExtensions.cs:228]: I will add a test case using:
- Plotly.NET.Interactive contains Plotly.NET.

File: src/Microsoft.DotNet.Interactive.PackageManagement/KernelSupportsNugetExtensions.cs
Comment (jonsequitur) [inline src/Microsoft.DotNet.Interactive.PackageManagement/KernelSupportsNugetExtensions.cs:216]: It looks like cancellation was never supported here. It would be great to fix this but I don't want to hold up the PR for it.

**Review by jonsequitur — APPROVED** [review]

File: src/Microsoft.DotNet.Interactive.PackageManagement/KernelSupportsNugetExtensions.cs
Comment (KevinRansom) [reply src/Microsoft.DotNet.Interactive.PackageManagement/KernelSupportsNugetExtensions.cs:216]: I'll add an issue

Comment (KevinRansom) [reply src/Microsoft.DotNet.Interactive.PackageManagement/KernelSupportsNugetExtensions.cs:216]: https://github.com/dotnet/interactive/issues/1515

**Review by jonsequitur — CHANGES_REQUESTED** [review]
Please add some tests and a screenshot.

**Comment by KevinRansom** [comment]
This is what it used to do every submission:

image

With this change now we show this when a cell is executed with a new #r "nuget"

image

Subsequent submissions will not display loaded messages for already loaded packages

image

**Comment by KevinRansom** [comment]
When logging is enabled the Log will contain these lines for every submission containing #re "nuget ...

````
2021-07-30T02:02:48.2694910Z [00-57fc302d10034647b9d540695d2f2caa-d4798017e5400242-00] [DoNugetRestore]    
2021-07-30T02:02:48.2695501Z [00-57fc302d10034647b9d540695d2f2caa-d4798017e5400242-00]   .NET: DisplayedValueProduced: Microsoft.DotNet.Interactive.InstallPackagesMessag ... 
2021-07-30T02:02:48.7760856Z [00-57fc302d10034647b9d540695d2f2caa-d4798017e5400242-00]   .NET: DisplayedValueUpdated: Microsoft.DotNet.Interactive.InstallPackagesMessag ... 
2021-07-30T02:02:49.2944290Z [00-57fc302d10034647b9d540695d2f2caa-d4798017e5400242-00]   .NET: DisplayedValueUpdated: Microsoft.DotNet.Interactive.InstallPackagesMessag ... 
2021-07-30T02:02:49.7993982Z [00-57fc302d10034647b9d540695d2f2caa-d4798017e5400242-00]   .NET: DisplayedValueUpdated: Microsoft.DotNet.Interactive.InstallPackagesMessag ... 
2021-07-30T02:02:50.0679326Z [00-57fc302d10034647b9d540695d2f2caa-d4798017e5400242-00] [DoNugetRestore]   (1798.4456ms) Installed Packages 
2021-07-30T02:02:50.0680065Z [00-57fc302d10034647b9d540695d2f2caa-d4798017e5400242-00] [DoNugetRestore]   (1798.5183ms)  - google.protobuf, 3.5.1 
2021-07-30T02:02:50.0680240Z [00-57fc302d10034647b9d540695d2f2caa-d4798017e5400242-00] [DoNugetRestore]   (1798.5357ms)  - microsoft.ml, 1.4.0 
2021-07-30T02:02:50.0680379Z [00-57fc302d10034647b9d540695d2f2caa-d4798017e5400242-00] [DoNugetRestore]   (1798.5496ms)  - microsoft.ml.cpumath, 1.4.0 
2021-07-30T02:02:50.0680484Z [00-57fc302d10034647b9d540695d2f2caa-d4798017e5400242-00] [DoNugetRestore]   (1798.56ms)  - microsoft.ml.dataview, 1.4.0 
2021-07-30T02:02:50.0680583Z [00-57fc302d10034647b9d540695d2f2caa-d4798017e5400242-00] [DoNugetRestore]   (1798.5699ms)  - microsoft.ml.onnxruntime, 0.5.1 
2021-07-30T02:02:50.0680695Z [00-57fc302d10034647b9d540695d2f2caa-d4798017e5400242-00] [DoNugetRestore]   (1798.5811ms)  - Microsoft.ML.OnnxTransformer, 1.4.0 
2021-07-30T02:02:50.0680796Z [00-57fc302d10034647b9d540695d2f2caa-d4798017e5400242-00] [DoNugetRestore]   (1798.5912ms)  - newtonsoft.json, 10.0.3 
2021-07-30T02:02:50.0680894Z [00-57fc302d10034647b9d540695d2f2caa-d4798017e5400242-00] [DoNugetRestore]   (1798.601ms)  - system.codedom, 4.4.0 
2021-07-30T02:02:50.0681000Z [00-57fc302d10034647b9d540695d2f2caa-d4798017e5400242-00] [DoNugetRestore]   (1798.6116ms)  - system.numerics.tensors, 0.1.0 
2021-07-30T02:02:50.0682283Z [00-57fc302d10034647b9d540695d2f2caa-d4798017e5400242-00]   .NET: DisplayedValueUpdated: Microsoft.DotNet.Interactive.InstallPackagesMessag ... 
2021-07-30T02:02:50.0684686Z [00-57fc302d10034647b9d540695d2f2caa-d4798017e5400242-00] [DoNugetRestore]   (1798.981ms)  
````

File: src/dotnet-interactive.IntegrationTests/LoggingTests.cs
Comment (jonsequitur) [inline src/dotnet-interactive.IntegrationTests/LoggingTests.cs:91]: Let's not integration test our logging. This is going to be fragile and non-critical.

Comment (KevinRansom) [reply src/dotnet-interactive.IntegrationTests/LoggingTests.cs:91]: How do I test logging spew?

Comment (jonsequitur) [reply src/dotnet-interactive.IntegrationTests/LoggingTests.cs:91]: We typically don't test diagnostic log output, though if it's worth doing, we can do it in-process in a more robust way. Let's chat about it later. I'd prefer to unblock this PR in the meantime.

**Review by colombod — APPROVED** [review]

**Pull request merged**
